fdbrpc/include/fdbrpc/LoadBalance.actor.h

```diff
@@ -713,6 +793,14 @@ Future<REPLY_TYPE(Request)> loadBalance(
 						}
 
 						if (firstRequestData.checkAndProcessResult(atMostOnce)) {
+							wait(firstRequestData.comparisonResult);
+							/*
```

Comment:
Is there any difference between doing a "wait()" on "comparisonResult" (as above) or doing a "wait()" on "replicaComparison()" (as below)? If we are doing a "wait()" on "comparisonResult" as above, and if "replicaComparison()" throws before the control reaches "wait()" would the error still be thrown?

Reply:
I think both would get the error. The error is not caught by this actor, thus will be rethrown.

`wait(firstRequestData.comparisonResult)` here is the correct. The below code starts another actor to do the comparison, which duplicates the work, since the work is already done in the actor for `firstRequestData.comparisonResult`.

Reply:
The same `wait` is needed in the other `return` cases above, i.e., checking the comparison results. So maybe refactor such a check before returning into a common function.

Reply:
> The below code starts another actor to do the comparison, which duplicates the work

I didn't explicitly state this but the idea is that if we are going to start another actor (like below) then we wouldn't invoke "maybeDoReplicaComparison()" at all.